src/main/java/ch/njol/skript/lang/function/ExprFunctionCall.java

```diff
@@ -90,15 +104,23 @@ public boolean areKeysRecommended() {
 	public <R> @Nullable Expression<? extends R> getConvertedExpression(Class<R>... to) {
 		if (CollectionUtils.containsSuperclass(to, getReturnType()))
 			return (Expression<? extends R>) this;
-		assert function.getReturnType() != null;
-		if (Converters.converterExists(function.getReturnType(), to))
-			return new ExprFunctionCall<>(function, to);
+
+		Class<?> returns = reference.signature().returnType();
+		Class<?> converterType;
+		if (returns != null && returns.isArray()) {
+			converterType = returns.componentType();
+		} else {
+			converterType = returns;
+		}
+
```

Comment:
this seems to be a common pattern, probably worth adding a helper/util method somewhere for it